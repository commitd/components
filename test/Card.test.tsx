import React from 'react'
import { renderLight, renderDark } from './setupTests'
import {
  Card,
  CardActionArea,
  CardMedia,
  CardHeader,
  CardContent,
} from '../src'

const CardExample = () => (
  <Card>
    <CardActionArea>
      <CardMedia
        height="120px"
        image="data:image/gif;base64,R0lGODlhEAAQAMQAAORHHOVSKudfOulrSOp3WOyDZu6QdvCchPGolfO0o/XBs/fNwfjZ0frl3/zy7////wAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAACH5BAkAABAALAAAAAAQABAAAAVVICSOZGlCQAosJ6mu7fiyZeKqNKToQGDsM8hBADgUXoGAiqhSvp5QAnQKGIgUhwFUYLCVDFCrKUE1lBavAViFIDlTImbKC5Gm2hB0SlBCBMQiB0UjIQA7"
        title="Committed image"
      />
      <CardHeader>Full Card Action</CardHeader>
      <CardContent>
        <p>The whole card can be an action area</p>
      </CardContent>
    </CardActionArea>
  </Card>
)

it('renders light without crashing', () => {
  const { asFragment } = renderLight(<CardExample />)
  expect(asFragment()).toMatchSnapshot()
})

it('renders dark without crashing', () => {
  const { asFragment } = renderDark(<CardExample />)
  expect(asFragment()).toMatchSnapshot()
})
