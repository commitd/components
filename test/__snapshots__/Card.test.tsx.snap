// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders dark without crashing 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiCard-root ForwardRef(withBox)-root ForwardRef(withBox)-root MuiPaper-elevation1 MuiPaper-rounded"
  >
    <button
      class="MuiButtonBase-root MuiCardActionArea-root"
      tabindex="0"
      type="button"
    >
      <div
        class="MuiCardMedia-root ForwardRef(withSizing)-root ForwardRef(withSizing)-root"
        style="background-image: url(data:image/gif;base64,R0lGODlhEAAQAMQAAORHHOVSKudfOulrSOp3WOyDZu6QdvCchPGolfO0o/XBs/fNwfjZ0frl3/zy7////wAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAACH5BAkAABAALAAAAAAQABAAAAVVICSOZGlCQAosJ6mu7fiyZeKqNKToQGDsM8hBADgUXoGAiqhSvp5QAnQKGIgUhwFUYLCVDFCrKUE1lBavAViFIDlTImbKC5Gm2hB0SlBCBMQiB0UjIQA7);"
        title="Committed image"
      />
      <div
        class="MuiCardHeader-root WithStyles(ForwardRef(CardHeader))-root WithStyles(ForwardRef(CardHeader))-root"
      >
        <div
          class="MuiCardHeader-content"
        >
          <span
            class="MuiTypography-root MuiCardHeader-title MuiTypography-h5 MuiTypography-displayBlock"
          >
            Full Card Action
          </span>
        </div>
      </div>
      <div
        class="MuiCardContent-root ForwardRef(withSizing)-root ForwardRef(withSizing)-root"
      >
        <p>
          The whole card can be an action area
        </p>
      </div>
      <span
        class="MuiCardActionArea-focusHighlight"
      />
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</DocumentFragment>
`;

exports[`renders light without crashing 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiCard-root ForwardRef(withBox)-root ForwardRef(withBox)-root MuiPaper-elevation1 MuiPaper-rounded"
  >
    <button
      class="MuiButtonBase-root MuiCardActionArea-root"
      tabindex="0"
      type="button"
    >
      <div
        class="MuiCardMedia-root ForwardRef(withSizing)-root ForwardRef(withSizing)-root"
        style="background-image: url(data:image/gif;base64,R0lGODlhEAAQAMQAAORHHOVSKudfOulrSOp3WOyDZu6QdvCchPGolfO0o/XBs/fNwfjZ0frl3/zy7////wAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAACH5BAkAABAALAAAAAAQABAAAAVVICSOZGlCQAosJ6mu7fiyZeKqNKToQGDsM8hBADgUXoGAiqhSvp5QAnQKGIgUhwFUYLCVDFCrKUE1lBavAViFIDlTImbKC5Gm2hB0SlBCBMQiB0UjIQA7);"
        title="Committed image"
      />
      <div
        class="MuiCardHeader-root WithStyles(ForwardRef(CardHeader))-root WithStyles(ForwardRef(CardHeader))-root"
      >
        <div
          class="MuiCardHeader-content"
        >
          <span
            class="MuiTypography-root MuiCardHeader-title MuiTypography-h5 MuiTypography-displayBlock"
          >
            Full Card Action
          </span>
        </div>
      </div>
      <div
        class="MuiCardContent-root ForwardRef(withSizing)-root ForwardRef(withSizing)-root"
      >
        <p>
          The whole card can be an action area
        </p>
      </div>
      <span
        class="MuiCardActionArea-focusHighlight"
      />
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</DocumentFragment>
`;
